lib/pbench/common/logger.py

```diff
@@ -8,6 +9,28 @@
 from pbench.common.exceptions import BadConfig
 
 
+class HostnameFilter(logging.Filter):
+    """
+    This logging filter allows adding hostname to standard Python log format
+    strings, in the form `%(hostname)s`
+    """
+
+    def __init__(self, hostname: str = None):
+        """
+        Initialize the filter. By default, `%(hostname)s` will be expanded to
+        the platform hostname. This can be overridden by specifying an explicit
+        hostname value.
+
+        Args:
+            hostname: Hostname to report. Defaults to platform.node().
+        """
+        self.hostname = hostname if hostname else platform.node()
+
+    def filter(self, record: logging.LogRecord) -> bool:
+        record.hostname = self.hostname
+        return True
```

Comment:
No, because in principle we could have more than one filter, and any filter returning `False` stops the chain.

Filtering is a bit weird, actually; the `logging.Filter` "base class" is actually an *example* of filtering by logger name. LGTM wants me to call the superclass `__init__`, but that just adds instance properties we don't want (and won't use in our `filter` overload). It's a great concept, but I get the sense maybe it's not used a lot. 😆

Reply:
I should add here that filters can do either (or both, I suppose) of two things: they can supply record attributes (like I do here) which are substituted via the format string (`%(hostname)`), or they can *reject* a log record based on some attribute (which is what the example `logging.Filter` base class does). In the latter case, you'd return `False` to reject the record: the log record will be dropped without any additional filters being run. In the former case, we don't want to drop the record, and we want to allow additional filters (of either variety), so we always return `True`.